python/ray/tests/test_advanced_9.py

```diff
@@ -438,6 +437,42 @@ def f():
         assert ray.get(f.remote())
 
 
+def test_gcs_fd_usage(shutdown_only):
+    ray.init(
+        _system_config={
+            "prestart_worker_first_driver": False,
+            "enable_worker_prestart": False,
+        },
+    )
+    gcs_process = ray._private.worker._global_node.all_processes["gcs_server"][0]
+    gcs_process = psutil.Process(gcs_process.process.pid)
+    print("GCS connections", len(gcs_process.connections()))
+
+    @ray.remote(runtime_env={"env_vars": {"Hello": "World"}})
+    class A:
+        def f(self):
+            import os
+
+            return os.environ.get("Hello")
+
+    # In case there are still some pre-start workers, consume all of them
+    aa = [A.remote() for _ in range(32)]
+    for a in aa:
+        assert ray.get(a.f.remote()) == "World"
+    base_fd_num = len(gcs_process.connections())
+    print("GCS connections", base_fd_num)
+
+    bb = [A.remote() for _ in range(4)]
+    for b in bb:
+        assert ray.get(b.f.remote()) == "World"
+    new_fd_num = len(gcs_process.connections())
+    print("GCS connections", new_fd_num)
+    # each worker has two connections:
```

Comment:
Hmm is this true (aren't they sharing the same fd)? I thought 2 connections are from 1 from python gcs client, and the other from core worker.

Reply:
No. I think one is from GCS -> CoreWorker (lease). The other one is CoreWorker -> GCS.

Reply:
(btw this is not a blocker or anything. Just would like to understand). IIUC, GCS->Core worker only happens when actors are scheduled. Is there any other known RPCs for GCS -> Core worker?

Reply:
@rkooo567 but the connection is not disconnected until the actor is dead. I think maybe we should put IDLE in the channel so it can disconnect automatically to save sockets.

Reply:
what's worse, we also have long polling for owner died :(